Controllers/HomeController.cs

```diff
@@ -8,100 +9,124 @@
             }
-            return View();//when submitted
+            
+            // Pass back the same Tour instance if the model is not valid.
+            return View(_repository.Tours.
+                        Where(t => t.TourId == tourId));//when submitted
         }
-        [HttpPost]
-        public IActionResult UserForm(TourParty tourParty)
+
+
+        [HttpGet]
+        public IActionResult UserForm(int tourId)
         {
-            if (ModelState.IsValid)//validates tour
+            _logger.LogInformation("GET request on UserForm.");
+            var signUp = new SignUpViewModel
             {
-                
-                //UnfinishedTour.AddTourist(tourParty);
-                //_repository.Tours.Where(t => t.TourId == UnfinishedTour.TourId);
+                TourId = tourId
+            };
 
-                //Product prodToUpdate = context.Products
-                //     .Where(p => p.ProductID == product.ProductID).FirstOrDefault();
+            // Return the ViewModel with the proper Tour object and a 
+            // new Party to be filled out.
+            return View(signUp);
+        }
 
-                //if (UnfinishedTour != null)
-                
-                Tour tour = _context.Tours.First(t => t.TourId == UnfinishedTour.TourId);
-                tour.AddTourist(tourParty);
+        [HttpPost]
+        public IActionResult UserForm(SignUpViewModel signUp)
+        {
+            _logger.LogInformation("POST request on UserForm.");
+            if (ModelState.IsValid)//validates tour
+            {
+                // If valid, add the Party to the DB.
+                Tour addTour = _repository.Tours
+                               .Where(t => t.TourId == signUp.TourId)
+                               .FirstOrDefault();
+                var party = new Party
+                {
+                    EmailAddress = signUp.EmailAddress,
+                    PartyName = signUp.PartyName,
+                    PhoneNumber = signUp.PhoneNumber,
+                    PartySize = signUp.PartySize,
+                    Tour = addTour,
+                    TourId = signUp.TourId
+                };
+                _context.Parties.Add(party);
+                addTour.Parties.Add(party);
                 _context.SaveChanges();
-               
-
-
-
-
-               
-
-
-                //save changes to db
                 return RedirectToAction("Index");//if valid returns UserForm
             }
-            return View();
-        }
-
-        [HttpGet]
-        public IActionResult UserForm(int tourId)
-        {
-            return View();
+            // If invalid, render the model with errors.
+            return View(signUp);
         }
 
         public IActionResult Appointments()
         {
             //this one should be ready
-            return View(_repository.Tours.OrderBy(t => t.TourTime));
+            return View(_repository.Tours
+                        .Include(t => t.Parties)
```

Comment:
.Include(t => t.Parties) is necessary. ASP.Net lazy loads by default, and this means that when passed to the View, Parties was empty. Due to null reference, the View was errorring out. This line tells ASP.Net to include the Parties objects when querying and sending Tours to the View.